depot/uploader/uploader.go

```diff
@@ -110,17 +111,15 @@ func (uploader *URLUploader) prepareFileForUpload(fileLocation string, logger la
 		return nil, 0, "", "", err
 	}
 
-	contents, err := io.ReadAll(sourceFile)
+	md5Checksum := md5.New()
+	sha256Checksum := sha256.New()
+	_, err = io.Copy(io.MultiWriter(md5Checksum, sha256Checksum), bufio.NewReader(sourceFile))
 	if err != nil {
 		logger.Error("failed-read", err)
```

Comment:
Perhaps we should return something like:

return nil, 0, "", "", err

when io.Copy fails as the error is only logged but not returned. It is better returning the error to ensure the caller is aware of the issue, isn't it?

Reply:
good catch, that was a simple typo :(
fixed